docs/pages/project/contributing/ci-build-and-release.md

```diff
@@ -81,6 +81,22 @@ Collectively, Meshery repositories will generally have CI workflow for commits a
 - Helm charts lint (helm)
 - Helm charts release, tag and push(stefanprodan/helm-gh-pages@master)
 
+### UI Build System
+
+Meshery UI (`/ui`) and Provider UI (`/provider-ui`) are built using [Next.js](https://nextjs.org/) with [SWC](https://swc.rs/) (Speedy Web Compiler) as the default compiler.
+
+**SWC Compiler Benefits**:
+- **Performance**: 20-70x faster compilation than Babel
+- **Built-in**: Default in Next.js 12+ (no additional configuration needed)
+- **Optimized**: Includes minification and Fast Refresh out of the box
+
+The Provider UI migrated from Babel to SWC to improve CI/CD build times and align with modern Next.js best practices. The migration removed Babel dependencies and configuration files (`.babelrc`), allowing Next.js to use its default SWC compiler.
```

Comment:
```suggestion
```

Thanks, @YASHMAHAKAL. This auxiliary information is not necessary to document here.